### ISSUE TITLE
"conan search --query" not working as documented

### ISSUE THREAD

**Issue by thorstenederer** [conan-io/conan#3547]
The latest [documentation](https://docs.conan.io/en/latest/reference/commands/consumer/search.html) states:

> If you specify a query filter for a setting and the package recipe is not restricted by this setting, will find all packages:

    class MyRecipe(ConanFile):
         settings="arch"

    $ conan search MyRecipe/1.0@lasote/stable -q os=Windows

> The query above will find all the MyRecipe binary packages, because the recipe doesn’t declare “os” as a setting.

I have created this MyRecipe package and searched for it in the local cache:

    $ conan search myrecipe/0.0.0@thorsten/testing

    Existing packages for recipe myrecipe/0.0.0@thorsten/testing:
    Package_ID: c0519e2d9702ec12d057bb15adb7a02baaf18107
        [settings]
            arch: x86_64
        Outdated from recipe: False

However, if I query for a setting which is irrelevant to the package, there is no match:

    $ conan search myrecipe/0.0.0@thorsten/testing -q os=Macos

    Existing packages for recipe myrecipe/0.0.0@thorsten/testing:
    There are no packages for reference 'myrecipe/0.0.0@thorsten/testing' matching the query 'os=Macos'

Only querying for os=None works:

    $ conan search myrecipe/0.0.0@thorsten/testing -q os=None
    
    Existing packages for recipe myrecipe/0.0.0@thorsten/testing:
    Package_ID: c0519e2d9702ec12d057bb15adb7a02baaf18107
        [settings]
            arch: x86_64
        Outdated from recipe: False

I favour the documented behavior over the evident behavior. Giving additional settings or options should not reject perfectly fine matches. In particular, with the current behavior, you cannot find matches for header-only libs (e.g. `boost_type_traits/1.66.0@bincrafters/stable`) when querying with _any_ setting.

Tested on: MacOS 10.13.6, conan 1.7.3

**Comment by lasote** [comment]
Hi, thanks for opening the issue. We changed that behavior because the `delete` is based on the search and it was very weird and dangerous to remove packages to say `-q os=Macos` and get deleted all the header only libraries. 
So we favored to keep the same behavior for removing and searching and we introduced the `None`. 
I opened a ticket in the docs to solve it ASAP: https://github.com/conan-io/docs/issues/823
